commit 19b9fc304f96e445b3584f3e80fcb515cec79c6a
Author: Vishnu Saini <vishsain@codeaurora.org>
Date:   Wed Dec 2 15:16:24 2020 +0530

    Xorg: Kde desktop bringup on RB5
    
    Fix glamor compilation issues.
    fallback glamor compositor to use fb_compositor
    as this is faster when glamor is not accelerated
    through GPU
    
    Signed-off-by: Vishnu Saini vishsain@codeaurora.org

diff --git a/glamor/glamor_egl.c b/glamor/glamor_egl.c
index 4bde637..6301973 100644
--- a/glamor/glamor_egl.c
+++ b/glamor/glamor_egl.c
@@ -808,7 +808,6 @@ glamor_egl_init(ScrnInfoPtr scrn, int fd)
 		goto error;  \
 	}
 
-    GLAMOR_CHECK_EGL_EXTENSION(MESA_drm_image);
     GLAMOR_CHECK_EGL_EXTENSION(KHR_gl_renderbuffer_image);
 #ifdef GLAMOR_GLES2
     GLAMOR_CHECK_EGL_EXTENSIONS(KHR_surfaceless_context, KHR_surfaceless_gles2);
diff --git a/glamor/glamor_render.c b/glamor/glamor_render.c
index a8b2081..503d26c 100644
--- a/glamor/glamor_render.c
+++ b/glamor/glamor_render.c
@@ -1590,7 +1590,7 @@ glamor_composite(CARD8 op,
     BoxPtr extent;
     int nbox, ok = FALSE;
     int force_clip = 0;
-
+    goto fail;
     if (source->pDrawable) {
         source_pixmap = glamor_get_drawable_pixmap(source->pDrawable);
         if (glamor_pixmap_drm_only(source_pixmap))
diff --git a/hw/xfree86/drivers/modesetting/drmmode_display.c b/hw/xfree86/drivers/modesetting/drmmode_display.c
index 45d5e90..a1dfaab 100644
--- a/hw/xfree86/drivers/modesetting/drmmode_display.c
+++ b/hw/xfree86/drivers/modesetting/drmmode_display.c
@@ -464,6 +464,7 @@ drmmode_ConvertToKMode(ScrnInfoPtr scrn,
     kmode->htotal = mode->HTotal;
     kmode->hskew = mode->HSkew;
 
+    kmode->vrefresh = mode->VRefresh;
     kmode->vdisplay = mode->VDisplay;
     kmode->vsync_start = mode->VSyncStart;
     kmode->vsync_end = mode->VSyncEnd;
@@ -1530,6 +1531,11 @@ drmmode_output_set_property(xf86OutputPtr output, Atom property,
     for (i = 0; i < drmmode_output->num_props; i++) {
         drmmode_prop_ptr p = &drmmode_output->props[i];
 
+        if (p->atoms == NULL) {
+            LogMessage(X_ERROR, "XorgLogL : p->atoms is NULL continuing\n");
+            continue;
+        }
+
         if (p->atoms[0] != property)
             continue;
 
